### ISSUE TITLE
EmptySource using mode RunsAndLumis ends prematurely

### ISSUE THREAD

**Issue by Dr15Jones** [cms-sw/cmssw#29796]
Using the example
```python
import FWCore.ParameterSet.Config as cms

process = cms.Process("Test")
process.source = cms.Source("EmptySource", numberEventsInLuminosityBlock = cms.untracked.uint32(2), processingMode=cms.untracked.string('RunsAndLumis'))

process.maxLuminosityBlocks.input = 3

process.add_(cms.Service("Tracer"))
```

stops processing after only handling 2 LuminosityBlocks instead of 3. It looks like the job always stops 1 Lumi to soon.

**Comment by Dr15Jones** [comment]
assign core

**Comment by Dr15Jones** [comment]
Fixed in #29804

**Comment by Dr15Jones** [comment]
+1
